DataAccess/DataAccessServiceRegistration.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using WebNetSample.DataAccess.Abstract;
+using WebNetSample.DataAccess.Concrete.EntityFramework;
+using WebNetSample.DataAccess.Concrete.EntityFramework.Contexts;
+
+namespace WebNetSample.DataAccess;
+
+public static class DataAccessServiceRegistration
+{
+    public static IServiceCollection AddDataAccessServices(this IServiceCollection services,
+                                                            IConfiguration configuration)
+    {
+        services.AddDbContext<WebNetSampleDBContext>(options =>
+                                                 options.UseSqlServer(
+                                                     configuration.GetConnectionString("WebNetSampleConnectionStringForWindows")));
+
+        services.AddSingleton<ISupplierRepository, EfSupplierRepository>();
+        services.AddSingleton<IProductRepository, EfProductRepository>();
+        services.AddSingleton<ICategoryRepository, EfCategoryRepository>();
+
+        return services;
```

Comment:
No need to return services since your method is an extension of functionality + it is a reference type, it will apply updates by reference, let's use `void` type for this method